sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -551,19 +596,20 @@ def gen_y_X_train_test(y, X, cv, cv_X):
 
     # final formatting of results DataFrame
     results = results.reset_index(drop=True)
-    if isinstance(scoring, List):
-        for s in scoring[1:]:
-            results[f"test_{s.name}"] = np.nan
-            for row in results.index:
-                results.loc[row, f"test_{s.name}"] = s(
-                    results["y_test"].loc[row],
-                    results["y_pred"].loc[row],
-                    y_train=results["y_train"].loc[row],
-                )
-
-    # drop pointer to data if not requested
-    if not return_data:
-        results = results.drop(columns=["y_train", "y_test", "y_pred"])
-    results = results.astype({"len_train_window": int})
 
+    # TODO: remove 13 lines below and 451-455 in v0.25.0
+    if raise_warn:
+        warnings.warn(
+            "Starting v0.25.0 all metric columns will be rearranged to the left of the "
```

Comment:
warnings should optimally always point out "where", e.g., "...in model_evaluation.evaluate ...", and optimally mention that it's about the output before you go into details.

(imagine the user who has a large module and gets 10 different warnings)

Reply:
I will mention 

> Starting v0.25.0 'evaluate' will be rearrange all metric columns to the left of the output result DataFrame. Please use loc references when addressing the columns.

is this ok? the users recognise what `evaluate` is if they use it in their program.

Reply:
> Starting v0.25.0 'evaluate' will rearrange all metric columns to the left of the output result DataFrame. Please use loc references when addressing the columns. You can safely ignore this warning if you don't use evaluate function directly.

should be clear enough to cover those two cases.

Reply:
ok, great! Perhaps mention `model_evaluation.evaluate`?

Reply:
sure of course! I updated the warning msg here 
62bdd51